### PR TITLE
Adds type definitions for the npm module "is-function"

### PR DESCRIPTION
**Pull request by evangeloszotos**
Please fill in this template.

- [ x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x] Test the change in your own code. (Compile and run.)
- [ x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Note:
`dts-gen --dt' => the setting  strictFunctionTypes was missing in the generated tsconfig. I added it myself. Just saying, maybe it could also be automatically placed there if it makes sense.

### CODE REVIEW

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/is-function/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for is-function 1.0
+// Project: https://github.com/grncdr/js-is-function
+// Definitions by: Evangelos Zotos <https://github.com/evangeloszotos>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function isFunction(fn: any): boolean;
+declare namespace isFunction {}
```
[the remark this answers is not in the document]
Comment (evangeloszotos) [reply types/is-function/index.d.ts]: great I had some reading on it

[the remark this answers is not in the document]
Comment (evangeloszotos) [reply types/is-function/index.d.ts]: i did fix it now i suppose :)

[the remark this answers is not in the document]
Comment (evangeloszotos) [reply types/is-function/index.d.ts]: I have fixed it. Now the tests use `import Foo from "foo"` and the index.d.ts does no tuse the namespace hack.

How can I trigger the review process of the pull request again?
I an new in contributing core to github projects, I am thankful for a little support :)

File: types/is-function/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// Type definitions for is-function 1.0
+// Project: https://github.com/grncdr/js-is-function
+// Definitions by: Evangelos Zotos <https://github.com/evangeloszotos>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export default function isFunction(fn: any): boolean;
```
Comment (plantain-00) [inline types/is-function/index.d.ts]: It's not `export default`: https://github.com/grncdr/js-is-function/blob/master/index.js#L1

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/is-function/index.d.ts (hunk 2 above)
Comment (evangeloszotos) [reply types/is-function/index.d.ts]: Oh my, yes, you are right, default export was the wrong approach, I spent now quite some time to figure out how to solve it best. 

if I delete the line with the namespace declaration:
`declare namespace isFunction {}`

then I it is message in visual studio code: 
image

I also tried to autogenerate the .d.ts file with dts-gen and here is what I received: 
image

All my search was bringing me back to that original proposal. 

I am open to a different approach, although I must confess I was able to solve it in a different way so far.

Comment (DanielRosenwasser) [reply types/is-function/index.d.ts]: You need to use `import isFunction = require("is-function")`. If you still want to use ES module imports, you need to enable `esModuleInterop`.

Comment (evangeloszotos) [reply types/is-function/index.d.ts]: I think know I have it right.

Thank you @DanielRosenwasser and @plantain-00 for your support on this

**Review by plantain-00 — APPROVED** [review]

**Pull request merged**